testbed/testbed/data_providers.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testbed
+
+import (
+	"encoding/binary"
+	"io"
+	"log"
+	"math/rand"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/data"
+	"go.opentelemetry.io/collector/internal/goldendataset"
+	"go.opentelemetry.io/collector/translator/internaldata"
+)
+
+type DataProvider interface {
+	GenerateTraces() pdata.Traces
+	GenerateTracesOld() []*tracepb.Span
+	GenerateMetrics() data.MetricData
+	GenerateMetricsOld() []*metricspb.Metric
+}
+
+type PerfTestDataProvider struct {
+	options       LoadOptions
+	batchesSent   *uint64
+	dataItemsSent *uint64
```

Comment:
Depending on what is the intent `GoldenDataProvider` could benefit from having these counters as well.

Nit: Perhaps also rename them from `*Sent` to `*Generated` or something like that since the data provider does not actually send the data, it only creates the data.

Reply:
Added to all DataProviders and renamed.